app/code/core/Mage/Adminhtml/Block/Widget/Grid/Column/Filter/Select.php

```diff
@@ -62,7 +62,7 @@ protected function _getOptions()
     protected function _renderOption($option, $value)
     {
         $selected = (($option['value'] == $value && (!is_null($value))) ? ' selected="selected"' : '');
-        return '<option value="' . $this->escapeHtml($option['value']) . '"' . $selected . '>' . $this->escapeHtml($option['label']) . '</option>';
+        return '<option value="' . $this->escapeHtml((string)$option['value']) . '"' . $selected . '>' . $this->escapeHtml($option['label']) . '</option>';
```

Comment:
```suggestion
        return '<option value="' . $this->escapeHtml($option['value']) . '"' . $selected . '>' . $this->escapeHtml($option['label']) . '</option>';
```
Revert after https://github.com/OpenMage/magento-lts/pull/2884